crates/trigger/src/cli.rs

```diff
@@ -147,29 +156,44 @@ where
         loader: impl Loader + Send + Sync + 'static,
         locked_url: String,
     ) -> Result<Executor> {
-        let trigger_config =
-            TriggerExecutorBuilderConfig::load_from_file(self.runtime_config_file.clone())?;
+        let runtime_config = self.build_runtime_config()?;
 
         let _sloth_warning = warn_if_wasm_build_slothful();
 
         let mut builder = TriggerExecutorBuilder::new(loader);
         self.update_wasmtime_config(builder.wasmtime_config_mut())?;
 
-        let logging_hooks = StdioLoggingTriggerHooks::new(self.follow_components(), self.log_dir());
+        if runtime_config.state_dir().is_none() {
```

Comment:
~Is there a particular reason to perform this check (and the `log_dir` one) in the trigger rather than in the `up` command?  Apart from the 'niceness' factor of having the UI stuff as far as possible on the `up` side, `up` has context for local vs remote and could skip the warning in cases where the user explicitly unset the state directory.~  Except this is a trigger-only option and `up` doesn't know about it Ivan